chain/bitcoind_client.go

```diff
@@ -36,99 +32,90 @@ type BitcoindClient struct {
 	started int32 // To be used atomically.
 	stopped int32 // To be used atomically.
 
-	client      *rpcclient.Client
-	connConfig  *rpcclient.ConnConfig // Work around unexported field
+	// birthday is the earliest time for which we should begin scanning the
+	// chain.
+	birthday time.Time
+
+	// chainParams are the parameters of the current chain this client is
+	// active under.
 	chainParams *chaincfg.Params
 
-	zmqConnect      string
-	zmqPollInterval time.Duration
+	// id is the unique ID of this client assigned by the backing bitcoind
+	// connection.
+	id uint64
 
-	notificationQueue *ConcurrentQueue
+	// chainConn is the backing client to our rescan client that contains
+	// the RPC and ZMQ connections to a bitcoind node.
+	chainConn *BitcoindConn
 
+	// bestBlock keeps track of the tip of the current best chain.
 	bestBlockMtx sync.RWMutex
 	bestBlock    waddrmgr.BlockStamp
 
-	clientMtx      sync.RWMutex
-	rescanUpdate   chan interface{}
-	startTime      time.Time
-	watchOutPoints map[wire.OutPoint]struct{}
-	watchAddrs     map[string]struct{}
-	watchTxIDs     map[chainhash.Hash]struct{}
-	notify         uint32
+	// notifyBlocks signals whether the client is sending block
+	// notifications to the caller.
+	notifyBlocks uint32
+
+	// rescanUpdate is a channel will be sent items that we should match
+	// transactions against while processing a chain rescan to determine if
+	// they are relevant to the client.
+	rescanUpdate chan interface{}
+
+	// watchedAddresses, watchedOutPoints, and watchedTxs are the set of
+	// items we should match transactions against while processing a chain
+	// rescan to determine if they are relevant to the client.
+	mu               sync.RWMutex
```

Comment:
nit: if this mustex is used exclusively for these maps, a better name would be `watchMtx` or similar